Admin/MonitorExchangeAuthCertificate/ConfigurationAction/Copy-ScriptToExchangeDirectory.ps1

```diff
@@ -25,14 +26,20 @@ function Copy-ScriptToExchangeDirectory {
         try {
             if (-not(Test-Path -Path $localScriptsPath)) {
                 Write-Verbose ("Folder: $($localScriptsPath) doesn't exist - it will be created now")
+                if ($WhatIfPreference) {
+                    Write-Host ("What if: Path '$($localScriptsPath)' doesn't exist and will now be created")
+                    $whatIfPathSimulation = $true
+                }
                 New-Item -Path $exchangeInstallPath -ItemType Directory -Name "Scripts" -ErrorAction Stop | Out-Null
             }
 
-            if (Test-Path -Path $localScriptsPath -ErrorAction Stop) {
+            if (($whatIfPathSimulation) -or
+                (Test-Path -Path $localScriptsPath -ErrorAction Stop)) {
                 Write-Verbose ("Path: $($localScriptsPath) was successfully created")
                 Copy-Item -Path $FullPathToScript -Destination $localScriptsPath -Force -ErrorAction Stop
```

Comment:
Same here. Do we inherit the setting?

Reply:
`Copy-Item` inherit the `$WhatIfPreference` and won't do the copy job then:
![image](https://user-images.githubusercontent.com/40993616/215836592-361a5bef-f54d-4da3-b3c1-05532f294422.png)